packages/desktop-client/src/components/mobile/transactions/TransactionList.jsx

```diff
@@ -119,6 +145,244 @@ export function TransactionList({
           );
         })}
       </ListBox>
+      {selectedTransactions?.length > 0 && (
+        <FloatingActionBar
+          transactions={transactions}
+          selectedTransactions={selectedTransactions}
+          onClearSelectedTransactions={onClearSelectedTransactions}
+          onEdit={onBatchEdit}
+          onDuplicate={onBatchDuplicate}
+          onLinkSchedule={onLinkSchedule}
+          onUnlinkSchedule={onUnlinkSchedule}
+          onSetTransfer={onSetTransfer}
+          onDelete={onBatchDelete}
+        />
+      )}
     </>
   );
 }
+
+function FloatingActionBar({
+  transactions,
+  selectedTransactions,
+  onClearSelectedTransactions,
+  onEdit,
+  onDuplicate,
+  onLinkSchedule,
+  onUnlinkSchedule,
+  onDelete,
+  style,
+}) {
+  const editMenuTriggerRef = useRef(null);
+  const [isEditMenuOpen, setIsEditMenuOpen] = useState(false);
+  const moreOptionsMenuTriggerRef = useRef(null);
+  const [isMoreOptionsMenuOpen, setIsMoreOptionsMenuOpen] = useState(false);
+  const getMenuItemStyle = item => ({
+    ...styles.mobileMenuItem,
+    color: theme.mobileHeaderText,
+    ...(item.name === 'delete' && { color: theme.errorTextMenu }),
+  });
+
+  const buttonProps = {
+    style: {
+      ...styles.mobileMenuItem,
+      color: 'currentColor',
+      height: styles.mobileMinHeight,
+    },
+    activeStyle: {
+      color: 'currentColor',
+    },
+    hoveredStyle: {
+      color: 'currentColor',
+    },
+  };
+
+  const allTransactionsAreLinked = useMemo(() => {
+    return transactions
+      .filter(t => selectedTransactions.includes(t.id))
+      .every(t => t.schedule);
+  }, [transactions, selectedTransactions]);
+
+  const isMoreThanOne = selectedTransactions.length > 1;
+
+  return (
+    <View
+      style={{
+        backgroundColor: theme.floatingActionBarBackground,
+        color: theme.floatingActionBarText,
+        position: 'fixed',
+        bottom: 10,
+        margin: '0 10px',
+        width: '95vw',
+        height: 60,
+        zIndex: 100,
+        borderRadius: 8,
+        border: `1px solid ${theme.floatingActionBarBorder}`,
+        ...style,
+      }}
+    >
+      <View
+        style={{
+          flex: 1,
+          padding: 8,
+          flexDirection: 'row',
+          alignItems: 'center',
+          justifyContent: 'space-between',
+        }}
+      >
+        <View
+          style={{
+            flexDirection: 'row',
+            alignItems: 'center',
+            justifyContent: 'flex-start',
+          }}
+        >
+          <Button
+            type="bare"
+            {...buttonProps}
+            style={{ ...buttonProps.style, marginRight: 4 }}
+            onClick={() => onClearSelectedTransactions?.()}
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** ‏What happens if `onClearSelectedTransactions` is undefined? The button is still visible, but nothing happens when clicked? That would be quite bad. Perhaps we can wrap the button in a conditional OR make the `onClearSelectedTransactions` prop mandatory?

Reply:
Updated. :) I'm thinking about whether we want to enable batch editing in the category transactions page. Thoughts?

Reply:
Ideally we would have equal feature-set in all the pages where we list transactions. So from that perspective - I think it makes sense to also enable in category transaction page.